bucket/drivers/s3/s3.go

```diff
@@ -0,0 +1,113 @@
+package s3
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/aws/aws-sdk-go/service/s3/s3manager"
+	"github.com/sirupsen/logrus"
+)
+
+type S3Driver struct {
+	//svc is IAM S3 service client
+	svc    *s3.S3
+	config *aws.Config
+}
+
+func New(config *aws.Config) *S3Driver {
+	sess, _ := session.NewSession(config)
+	// Create S3 service client
+	svc := s3.New(sess)
+	return &S3Driver{
+		svc:    svc,
+		config: config,
+	}
+}
+
+func (d *S3Driver) CreateBucket(name string) (string, error) {
+	_, err := d.svc.CreateBucket(&s3.CreateBucketInput{
+		Bucket: aws.String(name),
+		CreateBucketConfiguration: &s3.CreateBucketConfiguration{
+			LocationConstraint: aws.String(*d.config.Region),
+		},
+	})
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			switch aerr.Code() {
+			case s3.ErrCodeBucketAlreadyExists:
+				logrus.Error("Bucket name already in use!")
+				panic(err)
+			case s3.ErrCodeBucketAlreadyOwnedByYou:
+				logrus.Warn("Bucket exists and is owned by the requester")
+			default:
+				logrus.Error("Unknown error")
+			}
+		}
+		return "", err
+	}
+	logrus.Infof("Created Bucket %s in S3 ", name)
+	// Bucket name is same as bucket_id in S3
+	return name, nil
+}
+
+func (d *S3Driver) DeleteBucket(id string) error {
+	// DeleteBucket is possible only if the Bucket is empty.
```

Comment:
I think we should make this an option for the end-user. Let's add a bool parameter `clearBucket` and only clear the bucket if it's true. We can expose this as a BucketAccessClass option.

The end-user make not want to delete the bucket if it hasn't be cleared beforehand.